### PR TITLE
🤖 Put together a sketch at testing JATS

### PR DESCRIPTION
**Pull request by rowanc1**
`test`: test a JATS file against a list of unit tests in YAML

The test cases are useful for known exports and expecting specific pieces of information in the XML.

```bash
jats test article.jats --cases tests.yml
```

```yaml
cases:
  - title: Correct publisher ID (publisher-id)
    select: 'front > journal-meta > journal-id[journal-id-type="publisher-id"] > *'
    equals:
      type: text
      value: plos
  - title: Every orcid is authenticated
    selectAll: 'front > article-meta > contrib-group > contrib > contrib-id'
    equals:
      contrib-id-type: orcid
      authenticated: 'true'
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,6 +21,27 @@ jobs:
       - run: npm install
       - run: npm run lint:format
       - run: npm run lint
+  jats:
```
Comment (fwkoch) [inline .github/workflows/ci.yml:24]: Nice, would be good to have an equivalent to this on all our jats examples.

**Comment by rowanc1** [comment]
Going to merge this in and get it out the door, this could still change substantially as we start to learn things on which parts to test, etc.

File: src/utils.ts [hunk 2]
```diff
@@ -13,12 +13,20 @@ export function convertToUnist(node: Element): GenericNode | GenericParent | und
...
       return next;
     }
     case 'text': {
       const { attributes, text } = node;
-      return { type: 'text', ...attributes, value: String(text) };
+      return {
+        type: 'text',
+        ...attributes,
+        value: String(text)
+          .replace(/\n(\s+)$/, '')
```
Comment (fwkoch) [inline src/utils.ts:27]: Replacing these new lines is a little bit surprising, but I guess nothing really comes from these converted trees besides printing to the console, so there fewer lines is better.

**Review by fwkoch — APPROVED** [review]

File: src/utils.ts (hunk 2 above)
Comment (rowanc1) [reply src/utils.ts:27]: This is actually to do with formatting of the string in pandoc. They have a strange strategy for creating indented XML. This accommodates for this.

Most XML that I have seen from publishers is not indented, it is a single line.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rowanc1) [reply .github/workflows/ci.yml:24]: Yep, let's keep pushing on this here, and can bring it back into myst.

**Pull request merged**